vault/resource_pki_secret_backend_config_est.go

```diff
@@ -0,0 +1,181 @@
+)
+
+func pkiSecretBackendConfigEstResource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manages Vault PKI EST configuration",
+		CreateContext: provider.MountCreateContextWrapper(pkiSecretBackendConfigEstWrite, provider.VaultVersion116),
+		UpdateContext: pkiSecretBackendConfigEstWrite,
+		ReadContext:   pkiSecretBackendConfigEstRead,
+		DeleteContext: pkiSecretBackendConfigEstDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(_ context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				id := d.Id()
+				if id == "" {
+					return nil, fmt.Errorf("no path set for import, id=%q", id)
+				}
+
+				parts := strings.Split(mountutil.NormalizeMountPath(id), "/")
+				if err := d.Set("backend", parts[0]); err != nil {
+					return nil, err
+				}
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldBackend: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The PKI secret backend the resource belongs to.",
+				ForceNew:    true,
+			},
+			consts.FieldEnabled: {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Is the EST feature enabled",
+			},
+			consts.FieldDefaultMount: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Is this the cluster's default EST mount",
+			},
+			consts.FieldDefaultPathPolicy: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The behavior of the default_mount when enabled",
+			},
+			consts.FieldLabelToPathPolicy: {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "A pairing of EST label to the configured EST behavior for it",
+			},
+			consts.FieldAuthenticators: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Description: "Lists the mount accessors EST should delegate authentication requests towards",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+						"userpass": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+			consts.FieldEnableSentinelParsing: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Enable parsing of fields from the provided CSR for Sentinel policies",
+			},
+			consts.FieldAuditFields: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Fields parsed from the CSR that appear in the audit and can be used by sentinel policies",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func pkiSecretBackendConfigEstWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := verifyPkiEstFeatureSupported(meta); err != nil {
```

Comment:
This was added to verify that we were talking to an Enterprise version of Vault, but at the same time I also added the version. 

To my knowledge there isn't a way within `provider.MountCreateContextWrapper` to make it check for a version and the enterprise version?

Reply:
Ah, I see what you're saying here, @stevendpclark! I realize that ideally `provider.MountCreateContextWrapper` should also perform enterprise checks if needed. Since it may be out of scope for this PR, I'm fine moving ahead with the manual checks you have implemented here. I can add in an update in a follow-up PR to add the enterprise checks :)